core/drivers/scmi-msg/clock.c

```diff
@@ -321,7 +321,7 @@ static void scmi_clock_describe_rates(struct scmi_msg *msg)
 		}
 
 		out_count = rate_index - in_args->rate_index;
-		remaining = nb_rates - in_args->rate_index;
+		remaining = nb_rates - in_args->rate_index - out_count;
```

Comment:
`nb_rates - rate_index` would be better, no?

Reply:
I would have said `nb_rates - out_count` as out_count is the number of entries filled in the buffer. Maybe I'm missing something.

Reply:
> nb_rates - rate_index would be better, no?

Indeed simpler.

> I would have said `nb_rates - out_count` as out_count is the number of entries filled in the buffer. Maybe I'm missing something.

`nb_rates` gives the overall number of supported rates. `out_count` gives the number of rates filled in this very response message but does not count the rates skipped, from index 0 to index `in_args->rate_index - 1`.